### PR TITLE
docs: add vault helm enterprise examples

### PR DESCRIPTION
**Pull request by jasonodonnell**
This is the initial draft of the Vault Helm Enterprise example documentation for the upcoming Vault 1.4.0 release.  Includes applying license keys, setting up performance and disaster recovery clusters and setting up Raft storage.

### CODE REVIEW

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx [hunk 1]
```diff
@@ -0,0 +1,193 @@
+---
+layout: "docs"
+page_title: "Highly Available Vault Enterprise Disaster Recovery Clusters with Raft"
+sidebar_current: "docs-platform-k8s-examples-ha-enterprise-dr-with-raft"
+sidebar_title: "Highly Available Vault Enterprise Disaster Recovery Clusters with Raft"
+description: |-
+  Describes how to set up Diaster Recovery clusters with with Integrated Storage (Raft)
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:7]:
```suggestion
  Describes how to set up Diaster Recovery clusters with Integrated Storage (Raft)
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx [hunk 2]
```diff
@@ -0,0 +1,193 @@
+---
+layout: "docs"
+page_title: "Highly Available Vault Enterprise Disaster Recovery Clusters with Raft"
+sidebar_current: "docs-platform-k8s-examples-ha-enterprise-dr-with-raft"
+sidebar_title: "Highly Available Vault Enterprise Disaster Recovery Clusters with Raft"
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:5]: Might want to make the sidebar_title a little shorter? I remember the other example page titles getting pretty squished in the sidebar.

File: website/data/docs-navigation.js [hunk 3]
```diff
@@ -324,8 +324,12 @@ export default [
                   'standalone-tls',
                   'standalone-audit',
                   'ha-with-consul',
+                  'ha-with-raft',
                   'external',
                   'kubernetes-auth',
+                  'enterprise-with-raft.mdx',
+                  'enterprise-dr-with-raft.mdx',
+                  'enterprise-perf-with-raft.mdx',
```
Comment (tvoran) [inline website/data/docs-navigation.js:332]: Are these supposed to have a `.mdx` suffix? The netlify jobs seem to be failing with logs like this:
```
2:02:13 PM: Error: The page "enterprise-with-raft.mdx" was not found within the category 
"platform/k8s/helm/examples". Please double-check to ensure that 
"docs/platform/k8s/helm/examples/enterprise-with-raft.mdx.mdx" exists. If this page was never 
intended to exist, remove the key "enterprise-with-raft.mdx" from the category 
"platform/k8s/helm/examples" in "data/docs-navigation.js"
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+For more information on Disaster Recovery, [see the official documentation](docs/enterprise/replication/).
+
+## Primary Cluster
+
+First, create the primary cluster:
+
+```shell
+helm install vault-primary \
+  --set='server.image.repository=hashicorp/vault-enterprise' \
+  --set='server.image.tag=1.4.0-rc1_ent' \
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:25]: Will this still be `1.4.0-rc1_ent` after the release?
```suggestion
  --set='server.image.tag=1.4.0_ent' \
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+## Secondary Cluster
+
+With the primary cluster created, next create a secondary cluster and enable 
+disaster recovery replication.
+
+```shell
+helm install vault-secondary \
+  --set='server.image.repository=hashicorp/vault-enterprise' \
+  --set='server.image.tag=1.4.0-rc1_ent' \
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:78]:
```suggestion
  --set='server.image.tag=1.4.0_ent' \
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+```
+
+## Enable Disaster Recovery Replication On Primary
+
+With the initial clusters setup, we can now configure them for disaster recovery replication.
+
+First, on the primary cluster, enable replication:
+
+```shell
+kubectl exec -ti vault-primary-0 -- vault write -f sys/replication/dr/primary/enable primary_cluster_addr=https://vault-primary-active:8201
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:130]: Dumb question, but is replication always over `https`? Just noticing that `http` was used above for the `raft join` commands.

File: website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx [hunk 7]
```diff
@@ -0,0 +1,192 @@
+---
+layout: "docs"
+page_title: "Highly Available Vault Enterprise Performance Clusters with Raft"
+sidebar_current: "docs-platform-k8s-examples-ha-enterprise-perf-with-raft"
+sidebar_title: "Highly Available Vault Enterprise Performance Clusters with Raft"
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx:5]: Same comment about shortening the sidebar_title here.

File: website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+For more information on Disaster Recovery, [see the official documentation](docs/enterprise/replication/).
+
+## Primary Cluster
+
+First, create the primary cluster:
+
+```shell
+helm install vault-primary \
+  --set='server.image.repository=hashicorp/vault-enterprise' \
+  --set='server.image.tag=1.4.0-rc1_ent' \
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx:25]:
```suggestion
  --set='server.image.tag=1.4.0_ent' \
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+## Secondary Cluster
+
+With the primary cluster created, next create a secondary cluster.
+
+```shell
+helm install vault-secondary \
+  --set='server.image.repository=hashicorp/vault-enterprise' \
+  --set='server.image.tag=1.4.0-rc1_ent' \
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx:77]:
```suggestion
  --set='server.image.tag=1.4.0_ent' \
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-with-raft.mdx [hunk 10]
```diff
@@ -0,0 +1,97 @@
+---
+layout: "docs"
+page_title: "Highly Available Vault Enterprise Cluster with Raft"
+sidebar_current: "docs-platform-k8s-examples-ha-enterprise-with-raft"
+sidebar_title: "Highly Available Vault Enterprise Cluster with Raft"
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-with-raft.mdx:5]: Same comment about shortening here, something like this?
```suggestion
sidebar_title: "HA Enterprise Cluster with Raft"
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx [hunk 11]
```diff
@@ -0,0 +1,192 @@
+---
+layout: "docs"
+page_title: "Highly Available Vault Enterprise Performance Clusters with Raft"
+sidebar_current: "docs-platform-k8s-examples-ha-enterprise-perf-with-raft"
+sidebar_title: "Highly Available Vault Enterprise Performance Clusters with Raft"
+description: |-
+  Describes how to set up Performance clusters with with Integrated Storage (Raft)
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx:7]:
```suggestion
  Describes how to set up Performance clusters with Integrated Storage (Raft)
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-with-raft.mdx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+# Highly Available Vault Enterprise Cluster with Integrated Storage (Raft)
+
+~> **Important Note:** This chart is not compatible with Helm 2. Please use Helm 3 with this chart.
+
+Integrated storage (raft) can be enabled using the `server.ha.raft.enabled` value:
+
+```shell
+helm install vault \
+  --set='server.image.repository=hashicorp/vault-enterprise' \
+  --set='server.image.tag=1.4.0-rc1_ent' \
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-with-raft.mdx:19]:
```suggestion
  --set='server.image.tag=1.4.0_ent' \
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx (hunk 6 above)
Comment (jasonodonnell) [reply website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:130]: Yeah it is, I think Vault auto upgrades it.  I'll test and double check.

File: website/data/docs-navigation.js (hunk 3 above)
Comment (jasonodonnell) [reply website/data/docs-navigation.js:332]: Good catch, let me see if I can fix it.

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx (hunk 2 above)
Comment (jasonodonnell) [reply website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:5]: Yeah it's a little long, will shorten!

File: website/data/docs-navigation.js (hunk 3 above)
Comment (jasonodonnell) [reply website/data/docs-navigation.js]: Removed `.mdx`, hopefully it fixes it.

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx (hunk 2 above)
Comment (jasonodonnell) [reply website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx]: Done.

File: website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx [hunk 13]
```diff
@@ -0,0 +1,192 @@
+---
+layout: "docs"
+page_title: "Highly Available Vault Enterprise Performance Clusters with Raft"
+sidebar_current: "docs-platform-k8s-examples-ha-enterprise-perf-with-raft"
+sidebar_title: "HA Vault Enterprise Performance Clusters with Raft"
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-perf-with-raft.mdx:5]:
```suggestion
sidebar_title: "HA Enterprise Performance Clusters with Raft"
```

File: website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx [hunk 14]
```diff
@@ -0,0 +1,193 @@
+---
+layout: "docs"
+page_title: "Highly Available Vault Enterprise Disaster Recovery Clusters with Raft"
+sidebar_current: "docs-platform-k8s-examples-ha-enterprise-dr-with-raft"
+sidebar_title: "HA Vault Enterprise DR Clusters with Raft"
```
Comment (tvoran) [inline website/pages/docs/platform/k8s/helm/examples/enterprise-dr-with-raft.mdx:5]:
```suggestion
sidebar_title: "HA Enterprise DR Clusters with Raft"
```

**Pull request merged**